RecursiveExtractor/Extractors/SevenZipExtractor.cs

```diff
@@ -42,9 +42,44 @@ public async IAsyncEnumerable<FileEntry> ExtractAsync(FileEntry fileEntry, Extra
             {
                 Logger.Debug(Extractor.DEBUG_STRING, ArchiveFileType.P7ZIP, fileEntry.FullPath, string.Empty, e.GetType());
             }
+            var needsPassword = false;
```

Comment:
Generally this code is the same except the archive type.  Why not generalize and switch on type to avoid managing the same logic in a repeat method or at least making common the portion that could be like the password testing?

Reply:
Only the Rar and Sevenzip extractors share enough code (and a base type) to make this work.  But it would require casting the result and a potentially misleading API surface that takes any archive type.

I refactored so there's only one password checking section in each archive type.